app/src/main/java/com/example/development/sakaiclientandroid/NavActivity.java

```diff
@@ -201,6 +224,119 @@ public void onAllGradesFailure(Throwable t)
     }
 
 
+    /**
+     * Refreshes the site grades page by remaking the request for grades for that course
+     * @param siteId id of the course who's grades to refresh
+     * @param swipeRefreshLayout the swipe refresh layout from the site grades fragment
+     * @throws IllegalStateException if the current fragment isn't a site grade fragment, which shouldn't happen
+     */
+    public void refreshSiteGrades(String siteId, final SwipeRefreshLayout swipeRefreshLayout) throws IllegalStateException
```

Comment:
I like this feature. Is it possible to move it to the site pages fragment itself though, for separation of concerns?

Reply:
I put it there because i thought we wanted to put all calls to data handler's requests in NavActivity

Reply:
@akench Well that's not really necessary. DataHandler's purpose is to simply fetch data. I initially had the `loadAssignmentFragment` call moved to the activity so that we could just pass the assignments through the bundle (because of issues with the `treeview`); but if possible, it's almost always better to exercise tighter encapsulation. If this method can be moved to the fragment, it has a more specific context tied to it when someone tries to go back to read that code, too.